inc/compat/plugins/compat-plugin-woocommerce-gus.php

```diff
@@ -0,0 +1,388 @@
+		wp_register_script( 'frontend-ajax-gus', FluidCheckout_Enqueue::instance()->get_script_url( 'js/compat/plugins/woocommerce-gus/gus' ), array( 'jquery' ), NULL, array( 'in_footer' => true, 'strategy' => 'defer' ) );
+
+		// Add validation script
+		wp_register_script( 'fc-checkout-validation-woocommerce-gus', FluidCheckout_Enqueue::instance()->get_script_url( 'js/compat/plugins/woocommerce-gus/checkout-validation-woocommerce-gus' ), array( 'jquery', 'fc-utils', 'fc-checkout-validation' ), NULL, array( 'in_footer' => true, 'strategy' => 'defer' ) );
+		wp_add_inline_script( 'fc-checkout-validation-woocommerce-gus', 'window.addEventListener("load",function(){CheckoutValidationWooCommerceGUS.init(fcSettings.checkoutValidationWooCommerceGUS);})' );
+	}
+
+	/**
+	 * Enqueue scripts.
+	 */
+	public function enqueue_assets() {
+		// Scripts
+		wp_enqueue_script( 'fc-checkout-validation-woocommerce-gus' );
+	}
+
+	/**
+	 * Maybe enqueue assets.
+	 */
+	public function maybe_enqueue_assets() {
+		// Bail if not at checkout
+		if( ! FluidCheckout_Steps::instance()->is_checkout_page_or_fragment() ) { return; }
+
+		$this->enqueue_assets();
+	}
+
+
+
+	/**
+	 * Add settings to the plugin settings JS object.
+	 *
+	 * @param   array  $settings  JS settings object of the plugin.
+	 */
+	public function add_js_settings( $settings ) {
+		// Bail if plugin function is not available
+		if ( ! method_exists( 'wooGus\Gus\Connect', 'getOptions' ) ) { return $settings; }
+
+		// Add validation settings
+		$settings[ 'checkoutValidationWooCommerceGUS' ] = array(
+			'validationMessages'  => array(
+				'error_1' => __( 'Wtyczka nie została aktywowana', 'woogus'),
+				'error_2' => wooGus\Gus\Connect::getOptions( 'wprowadz_poprawny_nip' ),
+				'error_3' => __( 'Serwer GUS nie odpowiada, prosimy spróbować ponownie później', 'woogus'),
+				'to_short_nip' => __( 'Podany numer nip jest zbyt krótki', 'woogus'),
+			),
+		);
+
+		return $settings;
+	}
+
+
+
+	/**
+	 * Maybe change the VAT/NIP field arguments.
+	 *
+	 * @param  array  $field_groups   The checkout fields.
+	 */
+	public function maybe_change_vat_nip_field_properties( $field_groups ) {
+		// Bail if VAT field from the plugin is not available
+		if ( ! array_key_exists( 'billing', $field_groups ) || ! array_key_exists( self::VAT_FIELD_KEY, $field_groups[ 'billing' ] ) ) { return $field_groups; }
+
+		// Initalize variables
+		$field_args = $field_groups[ 'billing' ][ self::VAT_FIELD_KEY ];
+
+		// Set new priority
+		$field_args[ 'priority' ] = 230; // TODO: Remove this after fixing the fields display order caveat
+
+		// Set loading indicator on change
+		$class_args = array_merge( $field_args[ 'class' ], array( 'update_totals_on_change', 'loading_indicator_on_change' ) );
+		if ( class_exists( 'FluidCheckout_CheckoutFields' ) ) {
+			$class_args = FluidCheckout_CheckoutFields::instance()->merge_form_field_class_args( $field_args[ 'class' ], $class_args );
+		}
+		$field_args[ 'class' ] = $class_args;
```

Comment:
@vlad-dedikov,
- To trigger update on checkout fields, we can simply use the class `update_totals_on_change`.
- This will also set a delay to prevent multiple and too frequent requests while the customer is still typing;
- We can also show a loading indicator while updating, similar to how the account matching feature display it for the email field. To do this, we also set the class `loading_indicator_on_change`. This only works if the class `update_totals_on_change` is also used.